.github/workflows/checks.yml

```diff
@@ -0,0 +1,107 @@
+  pull_request:
+    branches: [ dev ]
+
+jobs:
+  Test:
+    name: Run linting and unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install requirements
+        run: |
+          pip install -r requirements.txt
+          pip install -r requirements_dev.txt
+      - name: Lint with flake8
+        run: flake8
+      - name: Calculate unit test coverage
+        run: |
+          coverage run --source foca -m pytest
+          coverage xml
+      - name: Submit Report to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          files: ./coverage.xml
+          fail_ci_if_error: true
+          verbose: true
+      - name: Run tests on petstore app
+        run: |
+          cd ./examples/petstore
+          docker-compose up --build -d
+          cd ../..
+          sleep 10
+          pytest ./tests/integration_tests.py
+
+  Docker:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version-tag: [["3.7", "3.7"], ["3.8", "3.8"], ["3.9", "3.9"], ["3.9", "latest"]]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+      - name: Build & Publish image to DockerHub
+        env:
+          docker_username: ${{ secrets.DOCKER_USERNAME }}
+          docker_password: ${{ secrets.DOCKER_PASSWORD }}
+          docker_org: ${{ secrets.DOCKER_ORG }}
+          repo_name: ${{ github.event.repository.name }}
+        run: |
+          export branch_name=${GITHUB_HEAD_REF##*/}
+          if [[ "$branch_name" == "dev" ]]; then
+              export tag="$(date '+%Y%m%d')"
+          else
+              export tag=${branch_name}
+          fi
+          if [[ ${{ matrix.python-version-tag[1] }} == "latest" ]]; then
+              export tag=${{ matrix.python-version-tag[1] }}
+          else
+              export tag=${tag}-py${{ matrix.python-version-tag[1] }}
+          fi
+          docker build . -t ${docker_org}/${repo_name}:latest \
+            -f docker/Dockerfile_py${{ matrix.python-version-tag[0] }}
+          docker tag ${docker_org}/${repo_name} ${docker_org}/${repo_name}:${tag}
+          echo $docker_password | docker login -u $docker_username --password-stdin
+          if [[ "$tag" == "latest" ]]; then
+              if [[ "$branch_name" == "dev" ]]; then
+                  docker push ${docker_org}/${repo_name}:${tag}
+              fi
+          else
+             docker push ${docker_org}/${repo_name}:${tag}
+          fi
+          rm ${HOME}/.docker/config.json
+
+  Release:
+    needs: [Test, Docker]
+    if: github.event_name == 'push' && github.ref == 'refs/heads/dev'
```

Comment:
I think through [its configuration](https://github.com/elixir-cloud-aai/foca/blob/71433c220c24bcb8071beccc57c89121d9eed16b/setup.cfg#L5-L14), Python Semantic Release is already set to only do anything when stuff is merged into the `master` branch, so it seems that this rule is actually conflicting. I think you can probably just delete the line. Python Semantic Release will run for every commit, but unless it's a push to the master branch, it will realize that there's nothing to do and quit gracefuly.